Client/Frontend/Fakespot/Components/CollapsibleCardContainer.swift

```diff
@@ -0,0 +1,169 @@
+
+import Common
+import UIKit
+
+class CollapsibleCardContainer: CardContainer, UIGestureRecognizerDelegate {
+    private struct UX {
+        static let verticalPadding: CGFloat = 8
+        static let horizontalPadding: CGFloat = 8
+        static let titleHorizontalPadding: CGFloat = 16
+        static let titleTopPadding: CGFloat = 16
+        static let chevronSize = CGSize(width: 20, height: 20)
+    }
+
+    enum ExpandButtonState {
+        case collapsed
+        case expanded
+
+        var image: UIImage? {
+            switch self {
+            case .expanded:
+                return UIImage(named: StandardImageIdentifiers.Large.chevronUp)?.withRenderingMode(.alwaysTemplate)
+            case .collapsed:
+                return UIImage(named: StandardImageIdentifiers.Large.chevronDown)?.withRenderingMode(.alwaysTemplate)
+            }
+        }
+
+        var toggle: ExpandButtonState {
+            switch self {
+            case .expanded:
+                return .collapsed
+            case .collapsed:
+                return .expanded
+            }
+        }
+    }
+
+    // MARK: - Properties
+    private var state: ExpandButtonState = .expanded
+
+    // UI
+    private lazy var rootView: UIView = .build { _ in }
+    private lazy var headerView: UIView = .build { _ in }
+    private lazy var containerView: UIView = .build { _ in }
+    private var containerHeightConstraint: NSLayoutConstraint?
+    private var tapRecognizer: UITapGestureRecognizer!
+
+    lazy var titleLabel: UILabel = .build { label in
+        label.adjustsFontForContentSizeCategory = true
+        label.font = DynamicFontHelper.defaultHelper.preferredFont(withTextStyle: .headline, size: 17.0)
+        label.numberOfLines = 0
+    }
+
+    private lazy var chevronButton: ResizableButton = .build { view in
+        view.setImage(self.state.image, for: .normal)
+        view.buttonEdgeSpacing = 0
+        view.addTarget(self, action: #selector(self.toggleExpand), for: .touchUpInside)
+    }
+
+    // MARK: - Inits
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        setupLayout()
+
+        tapRecognizer = UITapGestureRecognizer(target: self, action: #selector(tapHeader))
+        tapRecognizer.delegate = self
+        headerView.addGestureRecognizer(tapRecognizer)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func configure(_ view: UIView) {
+        configure(title: "", contentView: view, titleA11yId: "", closeButtonA11yId: "", expandState: .collapsed)
+    }
+
+    func configure(title: String,
+                   contentView: UIView,
+                   titleA11yId: String? = AccessibilityIdentifiers.Components.collapseButton,
+                   closeButtonA11yId: String? = AccessibilityIdentifiers.Components.cardTitleLabel,
+                   expandState: ExpandButtonState) {
```

Comment:
Just noticed, not sure if this is intended but a11yId for collapse and title are reversed.

Reply:
Good catch. I am about to open a PR for https://mozilla-hub.atlassian.net/browse/FXIOS-7091 and there these a11y ids are not used anymore so this won't be an issue.